core/cli/run.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+#include "cli/tree.hpp"
+
+namespace fc::cli {
+  inline bool isDash(const std::string &s) {
+    return !s.empty() && s[0] == '-';
+  }
+  inline bool isDashDash(const std::string &s) {
+    return s.size() == 2 && s[0] == '-' && s[1] == '-';
+  }
+  // note: returns first positional arg or end
+  inline Argv::iterator boostWorkaround(po::variables_map &vm,
+                                        const Opts &opts,
+                                        Argv::iterator begin,
+                                        Argv::iterator end) {
+    po::parsed_options parsed{&opts};
+    while (true) {
+      if (begin == end) {
+        break;
+      }
+      if (!isDash(*begin)) {
+        break;
+      }
+      if (isDashDash(*begin)) {
+        ++begin;
+        break;
+      }
+      const auto it{std::find_if(begin + 1, end, isDash)};
+      const auto options{
+          po::command_line_parser{Argv{begin, it}}.options(opts).run().options};
+      if (options.empty()) {
+        break;
+      }
+      for (const auto &option : options) {
+        parsed.options.emplace_back(option);
+        if (option.string_key.empty()) {
+          break;
+        }
```

Comment:
do we need this?

Reply:
yes, boost returns `option` for all passed argv, but positional args (and subcommands, which go after flags and their values) have empty `string_key`, so this ends flag parsing loop and leaves positional args untouched